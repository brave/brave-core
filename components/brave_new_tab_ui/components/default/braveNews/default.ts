// Copyright (c) 2020 The Brave Authors. All rights reserved.
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this file,
// You can obtain one at https://mozilla.org/MPL/2.0/.

import styled, { css, keyframes } from 'styled-components'

const blurOut = keyframes`
  from {
    filter: blur(20px);
  }
  to {
    filter: blur(0);
  }
`

// The page core structure
export const Section = styled('section')`
  z-index: 5;
  box-sizing: border-box;
  -webkit-font-smoothing: antialiased;
  min-height: 100vh;
  margin: 100px 30px;
  display: flex;
  align-items: center;
  flex-direction: column;
  transition: margin 2s ease-out;
  [data-show-news-prompt] & {
    margin-top: -100px;
  }
`

export const ArticlesGroup = styled('section')`
  box-sizing: border-box;
  filter: blur(20px);
  animation-name: ${blurOut};
  animation-duration: 0.7s;
  animation-timing-function: ease-out;
  animation-fill-mode: forwards;
`

// Generics

export const Block = styled('div')`
  overflow: hidden;
  box-sizing: border-box;
  position: relative;
  width: 680px;
  border-radius: 8px;
  padding: 36px 48px;
  background: rgba(53, 53, 53, 0.47);
  backdrop-filter: blur(62px);
  margin-bottom: 30px;
  color: white;
  a {
    text-decoration: none;
  }
`

export const Debugger = styled('h1')<{}>`
  background: red;
  position: absolute;
  padding: 5px 0;
  right: 0;
  top: 10px;
  text-transform: uppercase;
`

export const Image = styled('img')`
  box-sizing: border-box;
  max-width: 100%;
  display: block;
`

export const Text = styled('div')<{}>`
  box-sizing: border-box;
  font-family: ${p => p.theme.fontFamily.heading};
  font-weight: normal;
  font-size: 14px;
  line-height: 19px;
  color: #fff;
  margin: 0;
`

export const Heading = styled(Text.withComponent('h2'))`
  font-family: ${p => p.theme.fontFamily.heading};
  font-weight: 600;
  font-size: 22px;
  line-height: 32px;
`

export const Time = styled(Text.withComponent('time'))`
  box-sizing: border-box;
  display: block;
  font-family: ${p => p.theme.fontFamily.heading};
  font-size: 13px;
  margin-top: 4px;
  color: rgba(255,255,255,0.7);
`

export const PublisherLogo = styled('img')<{}>`
  box-sizing: border-box;
  max-width: 100%;
  height: 30px;
  min-width: 80px;
  background-color: rgba(188,188,188,0.2);
  margin-top: 12px;
  display: inline-block;
`

export const Button = styled('button')<{}>`
  appearance: none;
  cursor: pointer;
  display: block;
  border-radius: 24px;
  background: none;
  padding: 15px 34px;
  color: white;
  border: none;
  font-weight: 800;
  cursor: pointer;
  background: rgba(33, 37, 41, .8);
  backdrop-filter: blur(8px);
  outline: none;
  border: none;
  transition: opacity 1s ease-in-out, background .124s ease-in-out;
  &:hover {
    background: rgba(255, 255, 255, .2);
  }
  &:active {
    background: rgba(255, 255, 255, .4);
  }
  &:focus-visible {
    box-shadow: 0 0 0 1px ${p => p.theme.color.brandBrave};
  }
`

type CardButtonProps = {
  isMainFocus?: boolean
}
export const CardButton = styled(Button)<CardButtonProps>`
  backdrop-filter: none;
  background: rgba(255, 255, 255, .24);
  ${p => p.isMainFocus && css`
    display: block;
    width: 100%;
  `}
`

export const TertiaryButton = styled('button')<{}>`
  appearance: none;
  cursor: pointer;
  font-family: Poppins;
  font-weight: 600;
  font-size: 13px;
  background: none;
  border: none;
  outline: none;
  margin: 0;
  padding: 0;
  color: white;
  &:hover {
    color: #ddd;
  }
  &:active {
    transform: translate(1px, 1px)
  }
  &:focus-visible {
    box-shadow: 0 0 0 1px ${p => p.theme.color.brandBrave};
  }
`
