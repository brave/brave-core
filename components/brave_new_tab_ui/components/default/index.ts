/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at https://mozilla.org/MPL/2.0/. */

import { StatsContainer, StatsItem } from './stats'
export * from './settings'
import { AddSiteTile, AddSiteTileImage, Tile, TileActionsContainer, TileAction, TileFavicon, TileMenu, TileMenuItem, TileTitle } from './gridSites'
import { SiteRemovalNotification, SiteRemovalText, SiteRemovalAction } from './notification'
import { Clock } from './clock'
import { RewardsWidget } from './rewards'
import { BraveTalkWidget } from './braveTalk'
import EditCards from './editCards'
import EditTopSite from './editTopSite'
import SearchPromotion from './searchPromotion'
import createWidget from './widget'

export * from './page'

export {
  StatsContainer,
  StatsItem,
  AddSiteTile,
  AddSiteTileImage,
  Tile,
  TileActionsContainer,
  TileAction,
  TileFavicon,
  TileMenu,
  TileMenuItem,
  TileTitle,
  SiteRemovalNotification,
  SiteRemovalText,
  SiteRemovalAction,
  Clock,
  RewardsWidget,
  BraveTalkWidget,
  EditCards,
  createWidget,
  EditTopSite,
  SearchPromotion
}
