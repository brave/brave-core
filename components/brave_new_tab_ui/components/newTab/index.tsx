/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import * as React from 'react'
import { DragDropContext } from 'react-dnd'
import HTML5Backend from 'react-dnd-html5-backend'
import {
  Page,
  Header,
  ClockWidget as Clock,
  Main,
  ListWidget as List,
  Footer,
  DynamicBackground,
  Gradient
} from 'brave-ui/features/newTab/default'

// Components
import Settings from './settings'
import Stats from './stats'
import Block from './block'
import FooterInfo from './footerInfo'
import SiteRemovalNotification from './notification'

interface Props {
  newTabData: NewTab.State
  actions: any
  saveShowBackgroundImage: (value: boolean) => void
  saveShowClock: (value: boolean) => void
  saveShowTopSites: (value: boolean) => void
  saveShowStats: (value: boolean) => void
}

class NewTabPage extends React.Component<Props, {}> {
  componentDidMount () {
    // if a notification is open at component mounting time, close it
    this.props.actions.onHideSiteRemovalNotification()
  }

  onDraggedSite = (fromUrl: string, toUrl: string, dragRight: boolean) => {
    this.props.actions.siteDragged(fromUrl, toUrl, dragRight)
  }

  onDragEnd = (url: string, didDrop: boolean) => {
    this.props.actions.siteDragEnd(url, didDrop)
  }

  onToggleBookmark (site: NewTab.Site) {
    if (site.bookmarked === undefined) {
      this.props.actions.bookmarkAdded(site.url)
    } else {
      this.props.actions.bookmarkRemoved(site.url)
    }
  }

  onTogglePinnedTopSite (site: NewTab.Site) {
    if (!site.pinned) {
      this.props.actions.sitePinned(site.url)
    } else {
      this.props.actions.siteUnpinned(site.url)
    }
  }

  onIgnoredTopSite (site: NewTab.Site) {
    this.props.actions.siteIgnored(site.url)
  }

  toggleShowBackgroundImage = () => {
    this.props.saveShowBackgroundImage(
      !this.props.newTabData.showBackgroundImage
    )
  }

  toggleShowClock = () => {
    this.props.saveShowClock(
      !this.props.newTabData.showClock
    )
  }

  toggleShowStats = () => {
    this.props.saveShowStats(
      !this.props.newTabData.showStats
    )
  }

  toggleShowTopSites = () => {
    this.props.saveShowTopSites(
      !this.props.newTabData.showTopSites
    )
  }

  showSettings = () => {
    this.props.actions.showSettingsMenu()
  }

  closeSettings = () => {
    this.props.actions.closeSettingsMenu()
  }

  render () {
    const { newTabData, actions } = this.props

    if (!newTabData || !newTabData.backgroundImage) {
      return null
    }

    return (
      <DynamicBackground showBackgroundImage={newTabData.showBackgroundImage} background={newTabData.backgroundImage.source}>
        {newTabData.showBackgroundImage && <Gradient />}
        <Page>
          <Header>
            <Stats stats={newTabData.stats} showWidget={newTabData.showStats}/>
            <Clock showWidget={newTabData.showClock} />
            <Main>
              <List showWidget={newTabData.showTopSites}>
                {
                  this.props.newTabData.gridSites.map((site: NewTab.Site) =>
                    <Block
                      key={site.url}
                      id={site.url}
                      title={site.title}
                      href={site.url}
                      favicon={site.favicon}
                      style={{ backgroundColor: site.themeColor || site.computedThemeColor }}
                      onToggleBookmark={this.onToggleBookmark.bind(this, site)}
                      onPinnedTopSite={this.onTogglePinnedTopSite.bind(this, site)}
                      onIgnoredTopSite={this.onIgnoredTopSite.bind(this, site)}
                      onDraggedSite={this.onDraggedSite}
                      onDragEnd={this.onDragEnd}
                      isPinned={site.pinned}
                      isBookmarked={site.bookmarked !== undefined}
                    />
                  )
                }
              </List>
              {
                this.props.newTabData.showSiteRemovalNotification
                ? <SiteRemovalNotification actions={actions} />
                : null
              }
            </Main>
          </Header>
          {
            newTabData.showSettings &&
            <Settings
              onClickOutside={this.closeSettings}
              toggleShowBackgroundImage={this.toggleShowBackgroundImage}
              toggleShowClock={this.toggleShowClock}
              toggleShowStats={this.toggleShowStats}
              toggleShowTopSites={this.toggleShowTopSites}
              showBackgroundImage={newTabData.showBackgroundImage}
              showClock={newTabData.showClock}
              showStats={newTabData.showStats}
              showTopSites={newTabData.showTopSites}
            />
          }
          <Footer>
            <FooterInfo
              backgroundImageInfo={newTabData.backgroundImage}
              onClickSettings={this.showSettings}
              isSettingsMenuOpen={newTabData.showSettings}
              showPhotoInfo={newTabData.showBackgroundImage}
            />
          </Footer>
        </Page>
      </DynamicBackground>
    )
  }
}

export default DragDropContext(HTML5Backend)(NewTabPage)
