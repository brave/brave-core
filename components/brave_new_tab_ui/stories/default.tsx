/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import * as React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { storiesOf } from '@storybook/react'
import { withKnobs, select } from '@storybook/addon-knobs/react'
import Theme from 'brave-ui/theme/brave-default'
import DarkTheme from 'brave-ui/theme/brave-dark'
import BraveCoreThemeProvider from '../../common/BraveCoreThemeProvider'

// Components
import NewTabPage from '../containers/newTab'
import * as newTabActions from '../actions/new_tab_actions'
import * as gridSitesActions from '../actions/grid_sites_actions'
import * as rewardsActions from '../actions/rewards_actions'
import * as binanceActions from '../actions/binance_actions'
import * as geminiActions from '../actions/gemini_actions'
import store from '../store'
import { getNewTabData, getGridSitesData } from './default/data/storybookState'

export default function Provider ({ story }: any) {
  return (
    <ReduxProvider store={store}>
      <BraveCoreThemeProvider
        dark={DarkTheme}
        light={Theme}
        initialThemeType={select(
          'Theme',
          { ['Light']: 'Light', ['Dark']: 'Dark' },
          'Light'
        )}
      >
      {story}
      </BraveCoreThemeProvider>
    </ReduxProvider>
  )
}

storiesOf('New Tab/Containers', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <Provider story={story()} />)
  .add('Default', () => {
    const doNothing = (value: boolean) => value
    const newTabData = getNewTabData(store.getState().newTabData)
    const gridSitesData = getGridSitesData(store.getState().gridSitesData)
    return (
      <NewTabPage
        newTabData={newTabData}
        gridSitesData={gridSitesData}
        actions={Object.assign({}, newTabActions, gridSitesActions, rewardsActions, binanceActions, geminiActions)}
        saveShowBackgroundImage={doNothing}
        saveShowTopSites={doNothing}
        saveShowStats={doNothing}
        saveShowRewards={doNothing}
        saveShowBinance={doNothing}
        saveShowTogether={doNothing}
        saveShowAddCard={doNothing}
        saveShowGemini={doNothing}
        saveBrandedWallpaperOptIn={doNothing}
      />
    )
  })
