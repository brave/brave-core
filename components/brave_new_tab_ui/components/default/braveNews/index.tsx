// Copyright (c) 2020 The Brave Authors. All rights reserved.
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this file,
// You can obtain one at https://mozilla.org/MPL/2.0/.

import * as React from 'react'
import { useDispatch } from 'react-redux'
import * as BraveNews from '../../../../brave_news/browser/resources/shared/api'
import * as TodayActions from '../../../actions/today_actions'
import { useNewTabPref } from '../../../hooks/usePref'
import { defaultState } from '../../../storage/new_tab_storage'
import CardLoading from './cards/cardLoading'
import CardOptIn from './cards/cardOptIn'
import * as BraveNewsElement from './default'

const Content = React.lazy(() => import('./content'))
const FeedV2 = React.lazy(() => import('./FeedV2'))

// When FeedV2 is enabled, immediately start loading the Brave News chunk,
// rather than waiting for the parent component to render.
if (defaultState.featureFlagBraveNewsFeedV2Enabled) {
  import('./FeedV2')
}

export type OnReadFeedItem = (args: TodayActions.ReadFeedItemPayload) => any
export type OnSetPublisherPref = (publisherId: string, enabled: boolean) => any
export type OnPromotedItemViewed = (args: TodayActions.PromotedItemViewedPayload) => any
export type OnVisitDisplayAd = (args: TodayActions.VisitDisplayAdPayload) => any
export type OnViewedDisplayAd = (args: TodayActions.DisplayAdViewedPayload) => any
export type GetDisplayAdContent = BraveNews.BraveNewsControllerRemote['getDisplayAd']

export type Props = {
  isFetching: boolean
  hasInteracted: boolean
  isUpdateAvailable: boolean
  isPrompting: boolean
  feed?: BraveNews.Feed
  publishers?: BraveNews.Publishers
  articleToScrollTo?: BraveNews.FeedItemMetadata
  displayAdToScrollTo?: number
  displayedPageCount: number
  onReadFeedItem: OnReadFeedItem
  onPromotedItemViewed: OnPromotedItemViewed
  onVisitDisplayAd: OnVisitDisplayAd
  onViewedDisplayAd: OnViewedDisplayAd
  onFeedItemViewedCountChanged: (feedItemsViewed: number) => any
  onSetPublisherPref: OnSetPublisherPref
  onAnotherPageNeeded: () => any
  onCustomizeBraveNews: () => any
  onRefresh: () => any
  onCheckForUpdate: () => any
  getDisplayAd: GetDisplayAdContent
}

type PossibleInteractionObserver = IntersectionObserver | undefined

export const attributeNameCardCount = 'data-today-card-count'

const intersectionOptions = { root: null, rootMargin: '0px', threshold: 0.25 }

export default function BraveNewsSection(props: Props) {
  const dispatch = useDispatch()
  const [optedIn, setOptedIn] = useNewTabPref('isBraveNewsOptedIn')
  const [, setShowToday] = useNewTabPref('showToday')

  // Don't ask for initial data more than once
  const hasRequestedLoad = React.useRef(false)

  const loadDataObserver = React.useMemo<PossibleInteractionObserver>(() => {
    // Handle case where we're already told to load content, e.g. when
    // navigating back to a position in the feed after clicking an article
    // or ad.
    if (props.hasInteracted) {
      dispatch(TodayActions.refresh({ isFirstInteraction: false }))
      hasRequestedLoad.current = true
      return
    }
    // Handle case where we wait for News content area to hit viewport
    // before lazy-loading content.
    const handleHits: IntersectionObserverCallback = (entries) => {
      // When the scroll trigger hits the viewport (or is above the viewport),
      // we can start to fetch data.
      const isIntersecting = entries.some(entry => entry.isIntersecting)
      if (isIntersecting && !hasRequestedLoad.current) {
        // Only send interaction event when we're not loading data simply because
        // we're promoting items. In that case, the interaction event will
        // be sent by the content component when the content is first interacted
        // with.
        const shouldMarkInteraction = !props.isPrompting
        console.debug('Brave News: section is in position that requires requesting data load')
        dispatch(TodayActions.refresh({ isFirstInteraction: shouldMarkInteraction }))
        hasRequestedLoad.current = true
      }
    }
    // Setup intersection observer params
    return new IntersectionObserver(handleHits, intersectionOptions)
  }, [props.isPrompting, props.hasInteracted])

  const loadDataTrigger = React.useRef<HTMLDivElement>(null)

  React.useEffect(() => {
    // When we have an element to observe, set it as the target.
    // Don't do anything if we don't need data.
    if (!loadDataObserver ||
      !loadDataTrigger.current ||
      !optedIn ||
      !!props.feed) {
      return
    }
    loadDataObserver.observe(loadDataTrigger.current)
    return () => {
      // Cleanup current observer if we get a new observer, or a new element to observe
      loadDataObserver.disconnect()
    }
  }, [loadDataObserver, loadDataTrigger.current, optedIn, !!props.feed])

  // Only load all the content DOM elements if we're
  // scrolled far down enough, otherwise it's too easy to scroll down
  // by accident and get all the elements added.
  // Also sanity check isOptedIn, but without it there shouldn't be any content
  // anyway.
  const shouldDisplayContent = optedIn &&
    (props.hasInteracted || props.isPrompting)

  return (
    <BraveNewsElement.Section>
      <div
        ref={loadDataTrigger}
        style={{ position: 'sticky', top: '100px' }}
      />
      {!optedIn &&
        <>
          <CardOptIn onOptIn={() => setOptedIn(true)} onDisable={() => setShowToday(false)} />
        </>
      }
      {shouldDisplayContent && <React.Suspense fallback={defaultState.featureFlagBraveNewsFeedV2Enabled ? null : <CardLoading />}>
        {defaultState.featureFlagBraveNewsPromptEnabled
          && (defaultState.featureFlagBraveNewsFeedV2Enabled
            ? <FeedV2 />
            : <Content {...props} />)}
      </React.Suspense>}

    </BraveNewsElement.Section>
  )
}
