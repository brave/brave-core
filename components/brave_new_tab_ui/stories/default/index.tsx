/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import * as React from 'react'

// Feature-specific components
import { Page, Header, Main, Footer, DynamicBackground, Gradient, ClockWidget as Clock } from '../../components/default'

import TopSitesList from './topSites/topSitesList'
import Stats from './stats'
import SiteRemovalNotification from './siteRemovalNotification'
import FooterInfo from './footerInfo'

// Assets
import { getRandomBackgroundData } from './helpers'
import { images } from './data/background'

const generateRandomBackgroundData = getRandomBackgroundData(images)
interface State {
  showSettingsMenu: boolean
  showBackgroundImage: boolean
  showStats: boolean
  showClock: boolean
  showTopSites: boolean
}

interface Props {
  textDirection: string
}

export default class NewTabPage extends React.PureComponent<Props, State> {
  constructor (props: Props) {
    super(props)
    this.state = {
      showSettingsMenu: false,
      showBackgroundImage: true,
      showStats: true,
      showClock: true,
      showTopSites: true
    }
  }

  toggleShowBackgroundImage = () => {
    this.setState({ showBackgroundImage: !this.state.showBackgroundImage })
  }

  toggleShowClock = () => {
    this.setState({ showClock: !this.state.showClock })
  }

  toggleShowStats = () => {
    this.setState({ showStats: !this.state.showStats })
  }

  toggleShowTopSites = () => {
    this.setState({ showTopSites: !this.state.showTopSites })
  }

  closeSettings = () => {
    this.setState({ showSettingsMenu: false })
  }

  toggleSettings = () => {
    this.setState({ showSettingsMenu: !this.state.showSettingsMenu })
  }

  render () {
    const { showSettingsMenu, showBackgroundImage, showClock, showStats, showTopSites } = this.state
    const { textDirection } = this.props
    return (
      <DynamicBackground showBackgroundImage={showBackgroundImage} background={generateRandomBackgroundData.source} dir={textDirection}>
        {showBackgroundImage && <Gradient />}
        <Page>
          <Header>
            <Stats
              showWidget={showStats}
            />
            <Clock
              showWidget={showClock}
            />
            <Main>
              <TopSitesList
                showWidget={showTopSites}
              />
              <SiteRemovalNotification />
            </Main>
          </Header>
          <Footer>
            <FooterInfo
              textDirection={textDirection}
              onClickOutside={this.closeSettings}
              backgroundImageInfo={generateRandomBackgroundData}
              onClickSettings={this.toggleSettings}
              showSettingsMenu={showSettingsMenu}
              showPhotoInfo={showBackgroundImage}
              toggleShowBackgroundImage={this.toggleShowBackgroundImage}
              toggleShowClock={this.toggleShowClock}
              toggleShowStats={this.toggleShowStats}
              toggleShowTopSites={this.toggleShowTopSites}
              showBackgroundImage={showBackgroundImage}
              showClock={showClock}
              showStats={showStats}
              showTopSites={showTopSites}
            />
          </Footer>
        </Page>
      </DynamicBackground>
    )
  }
}
