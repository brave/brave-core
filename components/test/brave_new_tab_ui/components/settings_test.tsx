import * as React from 'react'
import { shallow } from 'enzyme'
import { SettingsMenu, SettingsWrapper } from 'brave-ui/features/newTab/default'
import Settings, { Props } from '../../../../components/brave_new_tab_ui/components/newTab/settings'

describe('settings component tests', () => {
  const mockProps: Props = {
    onClickOutside: () => null,
    toggleShowBackgroundImage: () => null,
    showBackgroundImage: true,
    toggleShowClock: () => null,
    toggleShowStats: () => null,
    toggleShowTopSites: () => null,
    showClock: true,
    showStats: true,
    showTopSites: true
  }

  it('should render the component properly', () => {
    const wrapper = shallow(
      <Settings
        onClickOutside={mockProps.onClickOutside}
        toggleShowBackgroundImage={mockProps.toggleShowBackgroundImage}
        showBackgroundImage={mockProps.showBackgroundImage}
        toggleShowClock={mockProps.toggleShowClock}
        toggleShowStats={mockProps.toggleShowStats}
        toggleShowTopSites={mockProps.toggleShowTopSites}
        showClock={mockProps.showClock}
        showStats={mockProps.showStats}
        showTopSites={mockProps.showTopSites}
      />)
    expect(wrapper.find(SettingsMenu)).toHaveLength(1)
    expect(wrapper.find(SettingsWrapper)).toHaveLength(1)
    expect(wrapper).toMatchSnapshot()
  })
})
