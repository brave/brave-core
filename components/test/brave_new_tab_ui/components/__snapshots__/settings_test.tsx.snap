// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings component tests should render the component properly 1`] = `ShallowWrapper {}`;
